src/Clockify/FollowUpService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Bot.Common;
+using Bot.States;
+using FluentDateTime;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Bot.Clockify
+{
+    public class FollowUpService : IFollowUpService
+    {
+        private readonly IUserProfilesProvider _userProfilesProvider;
+        private readonly string _microsoftAppId;
+        private readonly IClockifyMessageSource _messageSource;
+        private readonly IDateTimeProvider _dateTimeProvider;
+        private readonly ILogger<FollowUpService> _logger;
+
+        public FollowUpService(IUserProfilesProvider userProfilesProvider, IConfiguration configuration,
+            IClockifyMessageSource messageSource, IDateTimeProvider dateTimeProvider, ILogger<FollowUpService> logger)
+        {
+            _userProfilesProvider = userProfilesProvider;
+            _microsoftAppId = configuration["MicrosoftAppId"];
+            if (string.IsNullOrEmpty(_microsoftAppId))
```

Comment:
@khaelys do we really need to generate a dummy app id? The lack of the Microsoft app id in configuration is an error, isn't it? Wouldn't it be better to handle it here?

Reply:
create a separate issue though, let's not block this one

Reply:
When you start the bot locally you don't have an app id, that's why a random one is provided

Reply:
I'll add a comment on that section for clarification